ext/node/polyfills/process.ts

```diff
@@ -856,16 +853,6 @@ function synchronizeListeners() {
   } else {
     globalThis.removeEventListener("error", processOnError);
   }
-  if (beforeExitListenerCount > 0) {
-    globalThis.addEventListener("beforeunload", processOnBeforeUnload);
-  } else {
-    globalThis.removeEventListener("beforeunload", processOnBeforeUnload);
-  }
-  if (exitListenerCount > 0) {
-    globalThis.addEventListener("unload", processOnUnload);
-  } else {
-    globalThis.removeEventListener("unload", processOnUnload);
-  }
```

Comment:
Great, we should aim to remove remaining conditions from this function and get rid of it altogether (in separate PRs though)